src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -621,70 +716,68 @@ def _push(self) -> None:
         # entered_grid is zero at the end if a particle has never
         # entered any grid
         self.entered_grid += np.sum(self.particles_on_grid, axis=-1).astype(np.bool_)
+        # TODO: should this interpolation step be broken into a separate method
```

Comment:
No this is a great refactor! But add some comments explaining what's happening in each step a little more clearly to counteract the lower transparency of this compact phrase.